grails-app/services/org/pih/warehouse/core/DocumentTemplateService.groovy

```diff
@@ -39,6 +40,16 @@ class DocumentTemplateService {
         return output.toString()
     }
 
+    def renderInvoiceTemplate(Document documentTemplate, Shipment shipmentInstance, ByteArrayOutputStream outputStream) {
+        InputStream inputStream = new ByteArrayInputStream(documentTemplate.fileContents)
+        def requisitionNumber = shipmentInstance?.requisition?.requestNumber
```

Comment:
No need for this assignment, move it to `context.putVar("requisition", requisitionNumber ?: '')`

Reply:
ok